packaging/installer/kickstart.sh

```diff
@@ -317,73 +321,107 @@ NETDATA_INSTALLER_OPTIONS=""
+      fi
 
-    export NETDATA_LOCAL_TARBALL_OVERRIDE_CHECKSUM="${1}"
-    shift 1
+      export NETDATA_LOCAL_TARBALL_OVERRIDE_CHECKSUM="${3}"
 
-    if [ -z "${1}" ]; then
-      fatal "Missing go.d tarball: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
-    fi
+      if [ -z "${4}" ]; then
+        fatal "Missing go.d tarball: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
+      fi
 
-    export NETDATA_LOCAL_TARBALL_OVERRIDE_GO_PLUGIN="${1}"
-    shift 1
+      export NETDATA_LOCAL_TARBALL_OVERRIDE_GO_PLUGIN="${4}"
 
-    if [ -z "${1}" ]; then
-      fatal "Missing go.d config tarball: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
-    fi
+      if [ -z "${5}" ]; then
+        fatal "Missing go.d config tarball: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
+      fi
 
-    export NETDATA_LOCAL_TARBALL_OVERRIDE_GO_PLUGIN_CONFIG="${1}"
-    shift 1
+      export NETDATA_LOCAL_TARBALL_OVERRIDE_GO_PLUGIN_CONFIG="${5}"
 
-    if [ -z "${1}" ]; then
-      fatal "Missing dependencies management scriptlet: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
-    fi
+      if [ -z "${6}" ]; then
+        fatal "Missing dependencies management scriptlet: Option --local-files requires extra information. The desired tarball for netdata, the checksum, the go.d plugin tarball , the go.d plugin config tarball and the dependency management script, in this particular order"
+      fi
 
-    export NETDATA_LOCAL_TARBALL_OVERRIDE_DEPS_SCRIPT="${1}"
-    shift 1
-  else
-    NETDATA_INSTALLER_OPTIONS="$NETDATA_INSTALLER_OPTIONS ${1}"
-    shift 1
-  fi
+      export NETDATA_LOCAL_TARBALL_OVERRIDE_DEPS_SCRIPT="${6}"
+      shift 5
+      ;;
+    *)
+      NETDATA_INSTALLER_OPTIONS="${NETDATA_INSTALLER_OPTIONS} ${1}"
+      ;;
+  esac
+  shift 1
 done
 
 if [ "${INTERACTIVE}" = "0" ]; then
   PACKAGES_INSTALLER_OPTIONS="--dont-wait --non-interactive ${PACKAGES_INSTALLER_OPTIONS}"
   NETDATA_INSTALLER_OPTIONS="$NETDATA_INSTALLER_OPTIONS --dont-wait"
 fi
 
+if [ -n "${NETDATA_DISABLE_CLOUD}" ]; then
```

Comment:
Since the code is similar to the one in the `kickstart64-static.sh` script, we should share it from a common place.

Reply:
We can’t really share the code as the kickstart scripts need to be wholly self-contained.